weblate/checks/markup.py

```diff
@@ -256,6 +256,9 @@ def check_single(self, source, target, unit):
         if len(src_match) != len(tgt_match):
             return True
 
+        src_match.sort()
+        tgt_match.sort()
```

Comment:
I think it's already order independent - what is being compared are sets. Are these changes really useful?

Reply:
Using sets will miss if a URL is used more than once in a paragraph.  Using sorted lists will confirm that the source and translation have the same number of links as well as matching links.

Reply:
Oops, now I see what you're talking about, those vars are set comprehensions not list comprehensions.

Reply:
But you are right with missing the double usage, it's currently not detected.

Reply:
Actually double useage is not an issue either, because there's a length check before the parsed links get turned into a set.

Reply:
You're right!  Could you add a test of check failure so this is well documented for the future as a requirement?  Something like:
```python
        self.test_failure_1 = (
            "[Weblate](#weblate) has an [example]({{example}}) "
            "for illustrating the useage of [Weblate](#weblate)",
            "Ein [Beispiel]({{example}}) in [Webspät](#weblate) "
            "illustriert die Verwendung von [Webspät](#Webspät)",
            "md-text",
        )
        self.test_failure_2 = (
            "[Weblate](#weblate) has an [example]({{example}}) "
            "for illustrating the useage of [Weblate](#weblate)",
            "Ein [Beispiel]({{example}}) in [Webspät](#weblate) "
            "illustriert die Verwendung von Webspät",
            "md-text",
        )
```

Reply:
I've added the failure cases now.